### PR TITLE
Image refresh for rhel-9-0

### PR DESCRIPTION
**Pull request by cockpituous**
Image refresh for rhel-9-0
 * [x] image-refresh rhel-9-0
 * [x] https://github.com/cockpit-project/cockpit/pull/16236
 * [x] https://github.com/cockpit-project/cockpit/pull/16237

### CODE REVIEW

**Comment by allisonkarlitskaya** [comment]
Ugh.  Typical selinux-policy version mismatch trouble.

**Comment by cockpituous** [comment]
image-refresh in progress on 3-ci-srv-05.
Log: https://logs.cockpit-project.org/logs/image-refresh-2306-20210816-100808/

**Comment by cockpituous** [comment]
image-refresh rhel-9-0 done: https://github.com/cockpit-project/bots/commits/image-refresh-rhel-9-0-20210816-104343

**Comment by allisonkarlitskaya** [comment]
Twice unlucky?  Maybe it's time to port #2059 to rhel, after all.

**Comment by cockpituous** [comment]
image-refresh in progress on 4-cockpit-11.
Log: https://logs.cockpit-project.org/logs/image-refresh-2306-20210816-111414/

**Comment by cockpituous** [comment]
Task failed: https://logs.cockpit-project.org/logs/image-refresh-2306-20210816-111414/

File: images/scripts/rhel.setup [hunk 1]
```diff
@@ -296,6 +289,14 @@ EOF
     version=$(cut -d '-' -f 2 <<< "$IMAGE")
     su builder -c "/usr/bin/mock --verbose -i $(/var/lib/testvm/build-deps.sh "$opsys $version")"
     su builder -c "/usr/bin/mock --install --verbose rpmlint"
+
+    # We need to make sure mock has the same selinux-policy as we do
+    if [ "${IMAGE#rhel-9*}" != "$IMAGE" ]; then
+        dnf update -y selinux-policy
+        our_selinux_policy_version="$(rpm -q --qf '%{EVR}' selinux-policy)"
+        selinux_mock_packages="selinux-policy{,-devel,-targeted}-${our_selinux_policy_version}"
+        su builder -c "/usr/bin/mock --addrepo='$archive_repo' --install ${selinux_mock_packages}"
```
Comment (mvollmer) [inline images/scripts/rhel.setup:298]: `archive_repo` is not set, afaics.

**Comment by cockpituous** [comment]
image-refresh in progress on 1-cockpit-9.
Log: https://logs.cockpit-project.org/logs/image-refresh-2306-20210816-115757/

File: images/scripts/rhel.setup (hunk 1 above)
Comment (allisonkarlitskaya) [reply images/scripts/rhel.setup]: Ya.  That was a mistake.  Trying a different approach anyway.

File: images/scripts/rhel.setup [hunk 2]
```diff
@@ -296,6 +289,14 @@ EOF
     version=$(cut -d '-' -f 2 <<< "$IMAGE")
     su builder -c "/usr/bin/mock --verbose -i $(/var/lib/testvm/build-deps.sh "$opsys $version")"
     su builder -c "/usr/bin/mock --install --verbose rpmlint"
+
+    # We need to make sure mock has the same selinux-policy as we do
+    if [ "${IMAGE#rhel-9*}" != "$IMAGE" ]; then
+        mkdir /tmp/selinux-policy-rpms
+        dnf download --downloaddir /tmp/selinux-policy-rpms selinux-policy{,-devel,-targeted}
```
Comment (mvollmer) [inline images/scripts/rhel.setup:296]: Will this download the versions that are installed?  Or the ones that would be installed by `dnf update`?  Maybe we need the `dnf udpate selinux-blah` call after all.

File: images/scripts/rhel.setup [hunk 3]
```diff
@@ -296,6 +289,14 @@ EOF
...
+        dnf install /tmp/selinux-policy-rpms/*.rpm
+        su builder -c "/usr/bin/mock --install /tmp/selinux-policy-rpms/*.rpm"
```
Comment (mvollmer) [inline images/scripts/rhel.setup:298]: Do the host and mock share `/tmp`?  I guess the bots will tell us.  If not, there is `mock --copy-in`.

**Comment by cockpituous** [comment]
Task failed: https://logs.cockpit-project.org/logs/image-refresh-2306-20210816-115757/

File: images/scripts/rhel.setup (hunk 2 above)
Comment (mvollmer) [reply images/scripts/rhel.setup:296]: Arg, I missed the `dnf install` right after this, sorry.

File: images/scripts/rhel.setup [hunk 4]
```diff
@@ -296,6 +289,14 @@ EOF
     version=$(cut -d '-' -f 2 <<< "$IMAGE")
     su builder -c "/usr/bin/mock --verbose -i $(/var/lib/testvm/build-deps.sh "$opsys $version")"
     su builder -c "/usr/bin/mock --install --verbose rpmlint"
+
+    # We need to make sure mock has the same selinux-policy as we do
+    if [ "${IMAGE#rhel-9*}" != "$IMAGE" ]; then
+        mkdir /tmp/selinux-policy-rpms
+        dnf download --downloaddir /tmp/selinux-policy-rpms selinux-policy{,-devel,-targeted}
+        dnf install /tmp/selinux-policy-rpms/*.rpm
```
Comment (mvollmer) [inline images/scripts/rhel.setup:297]: `dnf install -y` :-)

**Comment by cockpituous** [comment]
image-refresh in progress on 2-cockpit-7.
Log: https://logs.cockpit-project.org/logs/image-refresh-2306-20210816-124545/

File: images/scripts/rhel.setup (hunk 3 above)
Comment (allisonkarlitskaya) [reply images/scripts/rhel.setup:298]: `--install` supports installing from local packages.

**Comment by cockpituous** [comment]
image-refresh rhel-9-0 done: https://github.com/cockpit-project/bots/commits/image-refresh-rhel-9-0-20210816-131616

**Comment by allisonkarlitskaya** [comment]
So quite a lot of stuff is failing.  One place this is happening is in `start_cockpit()` where it adds `notls.conf` with `ExecStartPre=-/bin/sh -c 'echo 0 > /proc/sys/kernel/yama/ptrace_scope'`.  In the podman tests, this entire block is failing, although it's not immediately clear why.

In any case, it results in:

```
Aug 16 09:59:43 localhost sh[1234]: /bin/sh: line 1: /proc/sys/kernel/yama/ptrace_scope: Read-only file system
```

which is weird, but (owing to `-`) doesn't prevent cockpit from starting.

**Comment by allisonkarlitskaya** [comment]
There's something very seriously wrong with `unix_chkpwd` on this image...

   1307 root      20   0    3924   2176   2020 R  99.3   0.2   0:20.91 unix_chkpwd                                      

It's causing cockpit to block on login, and affects password auth to ssh equally.

**Comment by allisonkarlitskaya** [comment]
It seems to be getting stuck in hashing the password.  That would also explain why creating the image pauses for a ridiculously long time when setting the password.

```
(gdb) bt
#0  0x00007fc2acaa0d2a in SHA512_Transform () from /lib64/libcrypt.so.2
#1  0x00007fc2acaa7b2e in _crypt_SHA512_Final () from /lib64/libcrypt.so.2
#2  0x00007fc2acaae6c1 in _crypt_crypt_sha512crypt_rn () from /lib64/libcrypt.so.2
#3  0x00007fc2acaabb85 in do_crypt () from /lib64/libcrypt.so.2
#4  0x00007fc2acaaf316 in crypt_r () from /lib64/libcrypt.so.2
#5  0x000055aeed6633a1 in main ()
```

And here's how `/etc/shadow` looks like:

```
admin:$6$rounds=999999999$EDbPvkiGrRHV9786$vdYPxuSCsb8CgsKL/UZ5obqm21LKowAQJN3DmMxWj3A8Jrpn63c9M2gp3S53FQ8ZFS2LH3/OljqNbw0sPWDDZ1:18855:0:99999:7:::
```

**Comment by cockpituous** [comment]
image-refresh in progress on 1-ci-srv-04.
Log: https://logs.cockpit-project.org/logs/image-refresh-2306-20210816-150909/

**Comment by cockpituous** [comment]
image-refresh rhel-9-0 done: https://github.com/cockpit-project/bots/commits/image-refresh-rhel-9-0-20210816-152424

**Comment by allisonkarlitskaya** [comment]
cockpit `TestAccounts.*` are failing because they use the system chpasswd command, which is bugged.

In any case, we can work around that with this entry from `/etc/login.defs`:

```
# If not specified, the libc will choose the default number of rounds (5000).
# The values must be within the 1000-999999999 range.
#
#SHA_CRYPT_MAX_ROUNDS 5000
```

which fixes the behaviour if we remove the `#`.

**Comment by cockpituous** [comment]
image-refresh in progress on 3-ci-srv-04.
Log: https://logs.cockpit-project.org/logs/image-refresh-2306-20210816-154848/

**Comment by cockpituous** [comment]
image-refresh rhel-9-0 done: https://github.com/cockpit-project/bots/commits/image-refresh-rhel-9-0-20210816-160404

**Comment by cockpituous** [comment]
image-refresh in progress on 1-ci-srv-04.
Log: https://logs.cockpit-project.org/logs/image-refresh-2306-20210816-160404/

**Comment by cockpituous** [comment]
image-refresh rhel-9-0 done: https://github.com/cockpit-project/bots/commits/image-refresh-rhel-9-0-20210816-161919

**Comment by marusak** [comment]
@allisonkarlitskaya you mentioned on irc that selinux is broken - it is rather simple one, just the string has changed.

**Review by mvollmer — APPROVED** [review]

**Comment by allisonkarlitskaya** [comment]
Here's the downstream bug for the password hashing issue, fwiw: https://bugzilla.redhat.com/show_bug.cgi?id=1993919

**Pull request merged**